nucypher/blockchain/eth/actors.py

```diff
@@ -656,3 +697,342 @@ def request_evaluation(self, evidence):
+        pass
+
+    def __init__(self,
+                 blockchain: BlockchainInterface,
+                 funding_account: str,
+                 funding_password: str = None,
+                 offline_mode: bool = False,
+                 sync_now: bool = True,
+                 *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        self.log = Logger(f"stakeholder")
+        self.blockchain = blockchain
+        self.offline_mode = offline_mode
+        self.eth_funding = Web3.toWei('0.1', 'ether')  # TODO: How much ETH to use while funding new stakes.
+
+        # Setup
+        if funding_account is not NO_FUNDING_ACCOUNT:
+            if not is_checksum_address(funding_account):
+                raise ValueError(f"{funding_account} is not a valid EIP-55 checksum address.")
+
+        self.funding_power = TransactingPower(blockchain=blockchain,
+                                              account=funding_account,
+                                              password=funding_password)
+        self.funding_power.activate()
+        self.__funding_account = funding_account
+
+        self.staking_agent = StakingEscrowAgent(blockchain=blockchain)
+        self.token_agent = NucypherTokenAgent(blockchain=blockchain)
+        self.economics = TokenEconomics()
+
+        self.__accounts = list()
+        self.__stakers = dict()
+        self.__transacting_powers = dict()
+
+        self.__get_accounts()
+
+        if sync_now and not offline_mode:
+            self.read_onchain_stakes()  # Stakes
+
+    def __str__(self) -> str:
+        r = f'{self.__class__.__name__}(funding_account={self.funding_account})'
+        return r
+
+    def __repr__(self) -> str:
+        r = f'{self.__class__.__name__}(funding_account={self.funding_account})'
+        return r
+
+    #
+    # Configuration
+    #
+
+    def static_payload(self) -> dict:
+        """Values to read/write from stakeholder JSON configuration files"""
+        payload = dict(provider_uri=self.blockchain.provider_uri,
+                       blockchain=self.blockchain.to_dict(),
+                       funding_account=self.funding_account,
+                       accounts=self.__accounts,
+                       stakers=self.__serialize_stakers())
+        return payload
+
+    @classmethod
+    def from_configuration_file(cls, filepath: str = None, **overrides) -> 'StakeHolder':
+        filepath = filepath or cls.default_filepath()
+        payload = cls._read_configuration_file(filepath=filepath)
+
+        # Sub config
+        blockchain_payload = payload.pop('blockchain')
+        blockchain = BlockchainInterface.from_dict(payload=blockchain_payload)
+        payload.update(dict(blockchain=blockchain))
+
+        payload.update(overrides)
+        instance = cls(filepath=filepath, **payload)
+        return instance
+
+    @validate_checksum_address
+    def attach_transacting_power(self, checksum_address: str, password: str = None) -> None:
+        try:
```

Comment:
I feel like the right way to do this is more like
```
if checksum_addresss in self.__transacting_powers:
    transacting_power = self.__transacting_powers[checksum_address]
else:
  transacting_power = TransactingPower(blockchain=self.blockchain, account=checksum_address)
  self.__transacting_powers[checksum_address] = transacting_power
transacting_power.activate(password=password)
```

I guess it's an ongoing debate...
https://softwareengineering.stackexchange.com/questions/351110/are-exceptions-for-flow-control-best-practice-in-python

Reply:
Heh - Yeah I even say that I prefer the exception handling as flow control, from the old idiom "Don't look before you leap"